project/scripts/build.bat

```diff
@@ -0,0 +1,458 @@
+@echo off
+setlocal enabledelayedexpansion
+
```

Comment:
My only concern with this file is that we need CI for windows. Related issue: #4588 

We can try to integrate `appveyor` and run `build.bat`, but NOT the whole test set.